### PR TITLE
[TECHNICAL] Unit tests for datasources classes - Part 2

### PR DESCRIPTION
**Pull request by Aitorbp**
## Related Issues
App:

- [x] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

### CODE REVIEW

**Review by manuelplazaspalacio — CHANGES_REQUESTED** [review]
Some changes here @Aitorbp

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 1]
```diff
@@ -43,10 +46,17 @@ import org.junit.Assert.assertNotNull
...
 
 class OCLocalFileDataSourceTest {
     private lateinit var localDataSource: OCLocalFileDataSource
     private lateinit var dao: FileDao
+    private val ocFileAndFileSync = OCFileAndFileSync(DUMMY_FILE_ENTITY, OCFileSyncEntity(
+        fileId = OC_FILE.id!!,
+        uploadWorkerUuid = null,
+        downloadWorkerUuid = null,
+        isSynchronizing = false
+    ), OC_SPACE_PERSONAL.toEntity())
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:59]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 2]
```diff
@@ -43,10 +46,17 @@ import org.junit.Assert.assertNotNull
...
+import java.util.UUID
 
 class OCLocalFileDataSourceTest {
     private lateinit var localDataSource: OCLocalFileDataSource
     private lateinit var dao: FileDao
+    private val ocFileAndFileSync = OCFileAndFileSync(DUMMY_FILE_ENTITY, OCFileSyncEntity(
+        fileId = OC_FILE.id!!,
+        uploadWorkerUuid = null,
+        downloadWorkerUuid = null,
+        isSynchronizing = false
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:58]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:59]: done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 3]
```diff
@@ -199,52 +268,430 @@ class OCLocalFileDataSourceTest {
...
+    @Test(expected = Exception::class)
+    fun `getFilesAvailableOfflineFromEveryAccount returns an exception when getFilesAvailableOfflineFromEveryAccount receive an exception`() {
+        every { dao.getFilesAvailableOfflineFromEveryAccount() } throws Exception()
+
+        localDataSource.getFilesAvailableOfflineFromEveryAccount()
+
+    }
+
+    @Test
+    fun `moveFile returns list of OCFile`() {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:478]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 4]
```diff
@@ -199,52 +268,430 @@ class OCLocalFileDataSourceTest {
...
+        val finalRemotePath = "/final/path"
+        val finalStoragePath = "final_storage"
+
+        every { dao.moveFile(any(), any(), finalRemotePath, finalStoragePath) } throws Exception()
+
+        localDataSource.moveFile(OC_FILE, OC_FILE_AVAILABLE_OFFLINE,finalRemotePath, finalStoragePath)
+    }
+
+    @Test
+    fun `saveFilesInFolderAndReturnThem returns list of OCFile`() {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:501]: done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 5]
```diff
@@ -199,52 +268,430 @@ class OCLocalFileDataSourceTest {
...
+    fun `saveFilesInFolderAndReturnThem returns exception when saveFilesInFolderAndReturnThem receive an exception`() {
+
+        every { dao.insertFilesInFolderAndReturnThem(any(), any() ) } throws Exception()
+
+        localDataSource.saveFilesInFolderAndReturnThem(listOf(OC_FILE), OC_FILE)
+
+    }
+
+    @Test
+    fun `saveFile returns unit`() {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:524]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 6]
```diff
@@ -199,52 +268,430 @@ class OCLocalFileDataSourceTest {
...
+    fun `saveFile returns an exception when saveFile receive an exception`() {
+
+        every { dao.upsert(any()) } throws Exception()
+
+        localDataSource.saveFile(OC_FILE)
+
+    }
+
+    @Test
+    fun `saveConflict returns unit`() {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:545]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 7]
```diff
@@ -199,52 +268,430 @@ class OCLocalFileDataSourceTest {
...
+
+        localDataSource.cleanConflict(OC_FILE.id!!)
+
+        verify (exactly = 1) {
+            dao.updateConflictStatusForFile(OC_FILE.id!!, null)
+        }
+    }
+
+    @Test(expected = Exception::class)
+    fun `cleanConflict returns exception`() {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:580]: done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 8]
```diff
@@ -199,52 +268,430 @@ class OCLocalFileDataSourceTest {
...
+
+        every { dao.updateConflictStatusForFile(any(), null) } throws Exception()
+
+        localDataSource.cleanConflict(OC_FILE.id!!)
+
+    }
+
+    @Test
+    fun `deleteFile returns unit`() {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:589]: done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 9]
```diff
@@ -199,52 +268,430 @@ class OCLocalFileDataSourceTest {
...
     @Test(expected = Exception::class)
-    fun `remove file - ko`() {
+    fun `deleteFile returns exception`() {
         every { dao.deleteFileById(any()) } throws Exception()
 
         localDataSource.deleteFile(DUMMY_FILE_ENTITY.id)
+    }
+
+    @Test
+    fun `deleteFilesForAccount returns unit`() {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:605]: done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 10]
```diff
@@ -199,52 +268,430 @@ class OCLocalFileDataSourceTest {
...
+        every { dao.deleteFilesForAccount(any()) } returns Unit
+
+        localDataSource.deleteFilesForAccount(DUMMY_FILE_ENTITY.name!!)
+
+        verify (exactly = 1) { dao.deleteFilesForAccount(DUMMY_FILE_ENTITY.name!!) }
+    }
+
+    @Test(expected = Exception::class)
+    fun `deleteFilesForAccount returns exception`() {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:614]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 11]
```diff
@@ -199,52 +268,430 @@ class OCLocalFileDataSourceTest {
...
+        every { dao.deleteFilesForAccount(any()) } throws Exception()
+
+        localDataSource.deleteFilesForAccount(DUMMY_FILE_ENTITY.name!!)
+    }
+
+    @Test
+    fun `disableThumbnailsForFile returns unit`() {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:621]: done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 12]
```diff
@@ -199,52 +268,430 @@ class OCLocalFileDataSourceTest {
...
+    @Test(expected = Exception::class)
+    fun `disableThumbnailsForFile returns exception`() {
+        every { dao.disableThumbnailsForFile(any()) } throws Exception()
 
-        verify { dao.deleteFileById(DUMMY_FILE_ENTITY.id) }
+        localDataSource.disableThumbnailsForFile(DUMMY_FILE_ENTITY.id)
     }
 
+    @Test
+    fun `updateAvailableOfflineStatusForFile returns unit`() {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:637]: done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 13]
```diff
@@ -199,52 +268,430 @@ class OCLocalFileDataSourceTest {
...
+
+        every { dao.updateAvailableOfflineStatusForFile(any(), any()) } returns Unit
+
+        localDataSource.updateAvailableOfflineStatusForFile(OC_FILE, newAvailableOfflineStatus)
+
+        verify (exactly = 1) { dao.updateAvailableOfflineStatusForFile(OC_FILE, newAvailableOfflineStatus.ordinal) }
+    }
+
+    @Test(expected = Exception::class)
+    fun `updateAvailableOfflineStatusForFile returns exception`() {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:648]: done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt [hunk 14]
```diff
@@ -199,52 +268,430 @@ class OCLocalFileDataSourceTest {
...
+
+        every { dao.updateSyncStatusForFile(any(), any()) } returns Unit
+
+        localDataSource.saveDownloadWorkerUuid(OC_FILE.id!!, workerUuid)
+
+        verify (exactly = 1) { dao.updateSyncStatusForFile(OC_FILE.id!!, workerUuid) }
+    }
+
+    @Test(expected = Exception::class)
+    fun `saveDownloadWorkerUuid returns exception`() {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt:668]: done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (manuelplazaspalacio) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt]: You forgot the last coma.

[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCLocalFileDataSourceTest.kt]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCRemoteFileDataSourceTest.kt [hunk 15]
```diff
@@ -198,6 +242,188 @@ class OCRemoteFileDataSourceTest {
...
+            OC_SPACE_PROJECT_WITH_IMAGE.webUrl,
+            true,
+        )
+
+        verify (exactly = 1) {
+            ocFileService.moveFile(
+                sourceRemotePath = any(),
+                targetRemotePath = OC_FILE.remotePath,
+                spaceWebDavUrl = OC_SPACE_PROJECT_WITH_IMAGE.webUrl,
+                replace = true
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCRemoteFileDataSourceTest.kt:280]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCRemoteFileDataSourceTest.kt [hunk 16]
```diff
@@ -198,6 +242,188 @@ class OCRemoteFileDataSourceTest {
...
+
+    @Test(expected = Exception::class)
+    fun `moveFile returns an exception when moveFile receive an exception`() {
+
+        every {
+            ocFileService.moveFile(
+                sourceRemotePath = any(),
+                targetRemotePath = OC_FILE.remotePath,
+                spaceWebDavUrl = OC_SPACE_PROJECT_WITH_IMAGE.webUrl,
+                replace = false
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCRemoteFileDataSourceTest.kt:310]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCRemoteFileDataSourceTest.kt [hunk 17]
```diff
@@ -198,6 +242,188 @@ class OCRemoteFileDataSourceTest {
...
+        assertEquals(expectedOCFile, result)
+
+        verify(exactly = 1) {
+            fileServiceMock.readFile( OC_FILE.remotePath, OC_SPACE_PROJECT_WITH_IMAGE.webUrl
+            )
+        }
+    }
+
+    @Test
+    fun `refreshFolder returns a list of OCFile`() {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCRemoteFileDataSourceTest.kt:356]: done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCRemoteFileDataSourceTest.kt [hunk 18]
```diff
@@ -198,6 +242,188 @@ class OCRemoteFileDataSourceTest {
...
 
+    @Test
+    fun `moveFile returns unit when replace is true`() {
+
+        every {
+            ocFileService.moveFile(
+                sourceRemotePath = any(),
+                targetRemotePath = OC_FILE.remotePath,
+                spaceWebDavUrl = OC_SPACE_PROJECT_WITH_IMAGE.webUrl,
+                replace = true
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCRemoteFileDataSourceTest.kt:263]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCRemoteFileDataSourceTest.kt [hunk 19]
```diff
@@ -198,6 +242,188 @@ class OCRemoteFileDataSourceTest {
...
+
+        val clientManagerMock = mockk<ClientManager>()
+
+        val remoteResult: RemoteOperationResult<RemoteFile> =
+            createRemoteOperationResultMock(data = remoteFile, isSuccess = true)
+
+        every {
+            fileServiceMock.readFile(
+                remotePath = OC_FILE.remotePath,
+                spaceWebDavUrl = OC_SPACE_PROJECT_WITH_IMAGE.webUrl
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCRemoteFileDataSourceTest.kt:338]: Done

File: owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCRemoteFileDataSourceTest.kt [hunk 20]
```diff
@@ -198,6 +242,188 @@ class OCRemoteFileDataSourceTest {
...
+            OC_FILE.remotePath,
+            OC_ACCOUNT_NAME,
+            OC_SPACE_PROJECT_WITH_IMAGE.webUrl,
+            false,
+        )
+
+    }
+
+    @Test
+    fun `readFile returns OCFile`() {
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply owncloudData/src/test/java/com/owncloud/android/data/file/datasources/OCRemoteFileDataSourceTest.kt:325]: Done

**Pull request merged**